Bounty - Medichem Goggles and Glasses

RedstonewolfX opened the pull request:
<!--
SPDX-FileCopyrightText: 2021 Pieter-Jan Briers <pieterjan.briers+git@gmail.com>
SPDX-FileCopyrightText: 2021 Swept <sweptwastaken@protonmail.com>
SPDX-FileCopyrightText: 2021 mirrorcult <lunarautomaton6@gmail.com>
SPDX-FileCopyrightText: 2022 AJCM-git <60196617+AJCM-git@users.noreply.github.com>
SPDX-FileCopyrightText: 2022 Kara <lunarautomaton6@gmail.com>
SPDX-FileCopyrightText: 2023 DrSmugleaf <DrSmugleaf@users.noreply.github.com>
SPDX-FileCopyrightText: 2023 Kevin Zheng <kevinz5000@gmail.com>
SPDX-FileCopyrightText: 2024 Vasilis <vasilis@pikachu.systems>
SPDX-FileCopyrightText: 2024 lzk <124214523+lzk228@users.noreply.github.com>
SPDX-FileCopyrightText: 2025 Aiden <28298836+Aidenkrz@users.noreply.github.com>

SPDX-License-Identifier: AGPL-3.0-or-later
-->

<!-- Guidelines: https://docs.spacestation14.io/en/getting-started/pr-guideline -->
<!-- NOTE: All code submitted to this repository is ALWAYS licensed under the AGPL-3.0-or-later license. 
The REUSE Specification headers or separate .license files indicate a secondary license (e.g., MPL or MIT) solely to facilitate 
integration for projects that do not use the AGPL license. This secondary license does not replace the fact that AGPL-3.0-or-later remains the primary and binding license. 
Uncomment and modify the following line if you wish to change the license from the default of AGPL.-->
<!--- LICENSE: AGPL -->
## About the PR
<!-- What did you change? -->
This PR adds a craftable version of Chemical Analysis Goggles that adds a medhud, called Medichem Goggles. They're crafted with any medhud, Chemical Analysis Goggles, a radio, and 5 LV wire, just like a medsec or meddiag HUD.

It also adds a set of Medichem Glasses, offering the medical HUD and solution viewing while also providing flash protection.
The Chief Medical Officer starts with these equipped, in place of their normal Medical Glasses.

## Why / Balance
<!-- Discuss how this would affect game balance or explain why it was changed. Link any relevant discussions or issues. -->
This is intended for Chemists to be able to have a medhud while still wearing chemical goggles.
The glasses are intended for the CMO.
Both of these items were created for a [bounty](https://discord.com/channels/1406768242843979816/1443280045643661382) by nightshadegale.

## Technical details
<!-- Summary of code changes for easier review. -->
Adds two new items, `ClothingEyesGlassesMedChem` and `ClothingEyesGlassesMedChemFlash`, and crafting recipes for them, `GogglesMedChem` and `GlassesMedChem`, respectively.
Adds a new `HudChemistry` tag to the Chemical Analysis Goggles, and a new `HudMedichem` tag to the Medichem Goggles, for crafting recipes.

## Media
<!-- Attach media if the PR makes ingame changes (clothing, items, features, etc).
Small fixes/refactors are exempt. Media may be used in SS14 progress reports with credit. -->
<img width="247" height="248" alt="image" src="https://github.com/user-attachments/assets/abb85bcf-0370-4d56-97ab-679e23256a30" />
<img width="282" height="265" alt="image" src="https://github.com/user-attachments/assets/83c3beaa-b83c-4d6e-b31b-b6925fea7490" />
<img width="294" height="301" alt="image" src="https://github.com/user-attachments/assets/7413e626-f1de-4c5a-b2fb-92096f40c517" />
<img width="291" height="209" alt="image" src="https://github.com/user-attachments/assets/4636f037-59e0-4119-95cb-5acf76b46d5f" />

## Requirements
<!-- Confirm the following by placing an X in the brackets [X]: -->
- [x] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [x] I have added media to this PR or it does not require an ingame showcase.
<!-- You should understand that not following the above may get your PR closed at maintainer’s discretion -->

## Breaking changes
<!-- List any breaking changes, including namespaces, public class/method/field changes, prototype renames; and provide instructions for fixing them.
This will be posted in #codebase-changes. -->

**Changelog**
<!-- Add a Changelog entry to make players aware of new features or changes that could affect gameplay.
Make sure to read the guidelines and take this Changelog template out of the comment block in order for it to show up.
Changelog must have a :cl: symbol, so the bot recognizes the changes and adds them to the game's changelog. -->
:cl:
- add: Medical Chemistry (Medichem) Goggles! Craftable like MedSec or MedDiag HUDs. Combines the features of a Medical HUD and Chemical Analysis Goggles.
- add: Medichem Glasses, craftable with Medichem Goggles, LV Cable, and Sunglasses.
- tweak: The Chief Medical Officer spawns with a free pair of Medichem Glasses.

RedstonewolfX committed:
Copy edits from Goobstation

Copying 3 commits I made in my goob fork

RedstonewolfX committed:
Implement and fix crafting recipes

Recipes weren't working but I managed to get them to work

RedstonewolfX committed:
Fix recipe wording

i may be stupid

RedstonewolfX committed:
Give CMO medichem glasses

The Chief Medical Officer gets Medichem Glasses equipped instead of Medical Glasses when they spawn.

Using a Chameleon Controller Implant gives you Medichem Glasses instead of a Medical HUD.

RedstonewolfX committed:
Trying to fix recipe

Why can't it just work man

RedstonewolfX committed:
Give Visiting CMO Medichem

RedstonewolfX committed:
Merge branch 'master' into medichem

RedstonewolfX committed:
Merge branch 'master' into medichem

dootythefrooty commented on Resources/Prototypes/Roles/Jobs/Fun/visitors_startinggear.yml:
```diff
@@ -76,6 +76,7 @@
   equipment:
     jumpsuit: ClothingUniformJumpsuitCMO
     shoes: ClothingShoesColorBrown
+    eyes: ClothingEyesGlassesMedChemFlash
```
```suggestion
    eyes: ClothingEyesGlassesMedChemFlash #omu
```

dootythefrooty commented on Resources/Prototypes/Roles/Jobs/Medical/chief_medical_officer.yml:
```diff
@@ -197,7 +197,7 @@
   equipment:
     head: ClothingHeadHatBeretCmo
-    eyes: ClothingEyesHudMedical
+    eyes: ClothingEyesGlassesMedChemFlash # Goobstation
```
```suggestion
    eyes: ClothingEyesGlassesMedChemFlash # Omu
```

dootythefrooty commented on Resources/Prototypes/Entities/Clothing/Eyes/glasses.yml:
```diff
@@ -448,6 +448,7 @@
       tags:
       - PetWearable
       - WhitelistChameleon
+      - HudChemistry
```
```suggestion
      - HudChemistry #omu edit
```

dootythefrooty reviewed (commented):
Pretty much everything needs to be namespaced to Omu, not Goobstation.

RedstonewolfX committed:
Update Namespaces

my bad

Co-authored-by: dootythefrooty <137359445+dootythefrooty@users.noreply.github.com>

CliveOcelot commented on Resources/Locale/en-US/_Goobstation/recipes/tags.ftl:
```diff
@@ -2,6 +2,8 @@ construction-graph-tag-beaker = beaker
 construction-graph-tag-medsec-hud = medsec hud
 construction-graph-tag-diagnostic-hud = diagnostic hud
+construction-graph-tag-chem-hud = chemical analysis goggles
+construction-graph-tag-medichem-hud = medichem goggles
```
put in omu namespace

CliveOcelot commented on Resources/Prototypes/_Goobstation/Recipes/Construction/Graphs/clothing/glasses_medichem.yml:
move to _Omu

CliveOcelot commented on Resources/Prototypes/_Goobstation/Entities/Clothing/Eyes/glasses.yml:
```diff
@@ -126,6 +126,43 @@
+  - type: Tag
+    tags:
+    - WhitelistChameleon
+
```
move to _Omu

CliveOcelot commented on Resources/Prototypes/_Goobstation/Recipes/Construction/Graphs/clothing/goggles_medichem.yml:
move to _Omu

RedstonewolfX committed:
Fix Namespaces Again

Who would have thought that the stuff I put in Omu would need to be in the Omu namespace

RedstonewolfX committed:
Minor spelling mistake

RedstonewolfX committed:
man

RedstonewolfX committed:
Merge branch 'ProjectOmu:master' into medichem

RedstonewolfX committed:
Merge branch 'master' into medichem

RedstonewolfX committed:
Merge branch 'master' into medichem

RedstonewolfX committed:
Merge branch 'master' into medichem

RichardBlonski committed:
Merge branch 'master' into medichem

RichardBlonski committed:
Removed Emtpy space

RichardBlonski reviewed (approved):
Works 👍
